PortalForecasts.R

```diff
@@ -144,20 +146,26 @@ forecastall <- function(abundances,level,weather_data,weatherforecast) {
   #Note: PI is missing. It has an error in the Poison Env model which produces NA values. 
   species=c('BA','DM','DO','DS','NA','OL','OT','PB','PE','PF','PH','PL','PM','PP','RF','RM','RO','SF','SH','SO','total')
   
+  #Model AIC sometimes doesn't work if species counts are low. In that case give a very large AIC.
   for(s in species) {
     species_abundance = abundances %>% 
       extract2(s)
 
     if(sum(species_abundance) == 0){
       pred = zero_abund_forecast
+      model_aic = 1e6
```

Comment:
This logic (predict zero_abund_forecast & set AIC to 1E6 if the predictions are sufficiently bad or if everyone is zero) seems to show up in multiple places as well. It would be good to have this logic live in one small function rather than getting repeated each time, I think